python/runfiles/runfiles.py

```diff
@@ -156,11 +190,13 @@ def Rlocation(self, path, source_repo=None):
         # target_repo is an apparent repository name. Look up the corresponding
         # canonical repository name with respect to the current repository,
         # identified by its canonical name.
-        target_canonical = self._repo_mapping[(source_repo, target_repo)]
-        return self._strategy.RlocationChecked(target_canonical + "/" + remainder)
+        if source_repo is not None:
```

Comment:
This `if` statement should always be `True` and it would be better to add an assert there. On line `180` there is
`if not remainder or (source_repo, target_repo) not in self._repo_mapping`. This means that if there is no remainder, then the code will return there, but if there is, then the `(None, target_repo) not in some_dict` should be always true because I would expect that value to never be present.

Reply:
I can promote this to an assertion